lib/WasmReader/WasmByteCodeGenerator.cpp

```
@@ -168,10 +168,19 @@ WasmBytecodeGenerator::GenerateFunction()
 
     WasmOp op;
     EmitInfo exprInfo;
-    while ((op = m_reader->ReadExpr()) != wnLIMIT)
+
+    try
+    {
+        while ((op = m_reader->ReadExpr()) != wnLIMIT)
+        {
+            exprInfo = EmitExpr(op);
+            ReleaseLocation(&exprInfo);
+        }
+    }
+    catch (WasmCompilationException& e)
```

I think we should catch all exception types, what do you think ?